core/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -273,6 +286,10 @@ private MatchingNodes findMatchingNodes(ShardRouting shard, RoutingAllocation al
             // we only check for NO, since if this node is THROTTLING and it has enough "same data"
             // then we will try and assign it next time
             Decision decision = allocation.deciders().canAllocate(shard, node, allocation);
```

Comment:
Same here about `explain` flag and allocation (it might be set higher up, I haven't checked yet, this is my first pass at this PR)